codex/slots/proofs/backends/asynccircoms.nim

```diff
@@ -0,0 +1,82 @@
+import std/options
+
+import pkg/taskpools
+import pkg/taskpools/flowvars
+import pkg/chronos
+import pkg/chronos/threadsync
+import pkg/questionable/results
+
+import ../../types
+import ../../../utils/asyncthreads
+
+import ./circomcompat
+
+type
+  AsyncCircomCompat* = object
+    params*: CircomCompatParams
+    tp*: Taskpool
+
+  # Args objects are missing seq[seq[byte]] field, to avoid unnecessary data copy
+  ProveTaskArgs* = object
+    signal: ThreadSignalPtr
+    params: CircomCompatParams
+
+var circomBackend {.threadvar.}: Option[CircomCompat]
```

Comment:
Each thread is gonna get it's own instance? I think the backend is already thread safe as it is?